modules/articles/client/config/articles.client.routes.js

```diff
@@ -12,22 +12,57 @@ angular.module('articles').config(['$stateProvider',
       })
       .state('articles.list', {
         url: '',
-        templateUrl: 'modules/articles/client/views/list-articles.client.view.html'
+        templateUrl: 'modules/articles/client/views/list-articles.client.view.html',
+        controller: 'ArticlesListController',
+        resolve:{
+          articles: function (Articles, $stateParams, $state) {
+            return Articles.query().$promise;
+          }
+        }
       })
       .state('articles.create', {
         url: '/create',
         templateUrl: 'modules/articles/client/views/create-article.client.view.html',
+        controller: 'ArticlesController',
         data: {
           roles: ['user', 'admin']
+        },
+        resolve: {
+          article: function () {
+            return { };
+          }
         }
       })
       .state('articles.view', {
         url: '/:articleId',
-        templateUrl: 'modules/articles/client/views/view-article.client.view.html'
+        templateUrl: 'modules/articles/client/views/view-article.client.view.html',
+        controller: 'ArticlesController',
+        resolve: {
+          article: function (Articles, $stateParams, $state) {
+            return Articles.get({
+              articleId: $stateParams.articleId
+            }).$promise;
+          }
+        }
       })
       .state('articles.edit', {
         url: '/:articleId/edit',
         templateUrl: 'modules/articles/client/views/edit-article.client.view.html',
+        controller: 'ArticlesController',
+        resolve: {
+          article: function (Articles, Authentication, $stateParams, $state) {
+            return Articles.get({
+              articleId: $stateParams.articleId
+            }).$promise.then(function (article) {
+                //Auth Check
+                if (article.user._id !== Authentication.user._id && Authentication.user.roles.indexOf('admin') === -1) {
+                  //TODO  Change to unauthorized when that PR is merged
+                  $state.go('unauthorized');
+                }
+                return article;
+              });
+          }
+        },
```

Comment:
Wait, how can the backend return something you're not authorised to see? You should get 403 in that case.

In any case, this is a lot of logic for route config. Thoughts?